src/terrain/material.hpp

```diff
@@ -93,10 +96,44 @@ class TerrainColorMapping {
         return colors_inverse_map;
     }
 
-    inline static unsigned int
+    inline static GLuint_p
     get_color_texture() {
         return color_texture_;
     };
+
+    inline static size_t
+    get_num_colors() {
+        return color_ids_map.size();
+    }
+};
+
+class MaterialGroup {
+ private:
+    std::map<MaterialId, std::set<ColorId>> allowable_;
+    std::set<MaterialId> allowable_materials_;
+
+ public:
+    MaterialGroup(){};
+    MaterialGroup(std::map<MaterialId, std::set<ColorId>> allowable,
+               std::set<MaterialId> allowable_materials) :
+        allowable_(allowable),
+        allowable_materials_(allowable_materials){};
+
+    [[nodiscard]] inline bool
+    material_in(MaterialId material_id, ColorId color_id) const {
+        if (material_in(material_id))
+            return true;
+        auto iter = allowable_.find(material_id);
+        if (iter != allowable_.end()) {
+            return iter->second.contains(color_id);
+        }
```

Comment:
this code is useless

by definition, it will only fun when `allowable_.find(material_id) == allowable_.end()`

Reply:
Nino, I would just like to thank you for spending the time to review my pull requests. That being said, please make your antecedents clear. It says `!=`. I have gotten this to run without a segfault.

Oh it's because my coding style gives psychic damage. Notice how one is allowable, and the other is allowable_materials.

Reply:
> Oh it's because my coding style gives psychic damage. Notice how one is allowable, and the other is allowable_materials.

then rename it

Reply:
Nino stop using pronouns. I don't understand what you are saying. I'll rename the member variables, but why should I remove it?

Reply:
Don't remove it

Rename the member variables

And add a comment about your psychic damage